src/Migrate/deploymentscripts/Initialize-AzMigrateReplicationInfrastructure.ps1

```diff
@@ -0,0 +1,386 @@
+                Write-Host $LogStringCreated, $CommonLockName, " for ", $ServiceBusNamespace
+            }else{
+                Write-Host $CommonLockName, " for ", $ServiceBusNamespace, $LogStringSkipping
+            }
+
+            $kvLock = Get-AzResourceLock -LockName $CommonLockName -ResourceName $KeyVaultName -ResourceType $KeyVaultType -ResourceGroupName $ResourceGroupName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$kvLock){
+                New-AzResourceLock -LockLevel CanNotDelete -LockName $CommonLockName -ResourceName $KeyVaultName -ResourceType $KeyVaultType -ResourceGroupName $ResourceGroupName -LockNotes $lockNotes -Force
+                Write-Host $LogStringCreated, $CommonLockName, " for ", $KeyVaultName
+            }else{
+                Write-Host $CommonLockName, " for ", $KeyVaultName, $LogStringSkipping
+            }
+            
+
+            # Intermediate phase
+            # RoleAssignments
+            
+            $roleDefinitionId = "81a9662b-bebf-436f-a333-f67b29880f12"
+            $kvspnid = (Get-AzADServicePrincipal -DisplayName "Azure Key Vault" )[0].Id
+            $gwyStorageAccount = Get-AzResource -ResourceName $GateWayStorageAcName -ResourceGroupName $ResourceGroupName 
+            $lsaStorageAccount = Get-AzResource -ResourceName $LogStorageAcName -ResourceGroupName $ResourceGroupName
+            $gwyRoleAssignments = Get-AzRoleAssignment -ObjectId $kvspnid -Scope $gwyStorageAccount.Id -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            $lsaRoleAssignments = Get-AzRoleAssignment -ObjectId $kvspnid -Scope $lsaStorageAccount.Id -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$lsaRoleAssignments){
+                New-AzRoleAssignment -ObjectId $kvspnid -Scope $lsaStorageAccount.Id -RoleDefinitionId $roleDefinitionId
+                Write-Host $LogStringCreated, "RoleAssignment", " for ", $LogStorageAcName
+            }else{
+                Write-Host "RoleAssignment", " for ", $LogStorageAcName, $LogStringSkipping
+            }
+            if(!$gwyRoleAssignments){
+                New-AzRoleAssignment -ObjectId $kvspnid -Scope $gwyStorageAccount.Id -RoleDefinitionId $roleDefinitionId
+                Write-Host $LogStringCreated, "RoleAssignment", " for ", $GateWayStorageAcName
+            }else{
+                Write-Host "RoleAssignment", " for ", $GateWayStorageAcName, $LogStringSkipping
+            }
+
+            # SA. SAS definition
+
+            $gatewayStorageAccountSasSecretName = "gwySas"
+            $cacheStorageAccountSasSecretName = "cacheSas"
+            $regenerationPeriod = [System.Timespan]::FromDays(30)
+            $keyName = 'Key2'
+            Add-AzKeyVaultManagedStorageAccount -VaultName $KeyVaultName -AccountName $LogStorageAcName -AccountResourceId  $lsaStorageAccount.Id  -ActiveKeyName $keyName -RegenerationPeriod $regenerationPeriod
```

Comment:
portal also uses --auto-regenerate-key to true. plz set this arg as well.

Reply:
In PS, by default auto regen. We have to disable if we want
ActiveKeyName      : key2                                                                                                          **AutoRegenerateKey  : True**                                                                                                          RegenerationPeriod : 30.00:00:00                                                                                                   
AccountName        : migrategwsa1092702858                                                                                         AccountResourceId  : /subscriptions/7c943c1b-5122-4097-90c8-861411bdd574/resourceGroups/azmigratepwshtestasr13072020/providers/M                        icrosoft.Storage/storageAccounts/migrategwsa1092702858                                                 8